src/main/java/nl/jqno/equalsverifier/internal/util/Configuration.java

```diff
@@ -79,90 +84,126 @@ public PrefabValues getPrefabValues() {
-                cachedHashCodeInitializer, hasRedefinedSuperclass, redefinedSubclass, usingGetClass, warningsToSuppress);
+        return new Configuration<>(type, prefabValues, value, unequalExamples, actualFields, excludedFields,
+                includedFields, nonnullFields, ignoredAnnotations, cachedHashCodeInitializer, hasRedefinedSuperclass,
+                redefinedSubclass, usingGetClass, warningsToSuppress);
     }
 
     public List<T> getEqualExamples() {
         return Collections.unmodifiableList(equalExamples);
     }
 
     public Configuration<T> withUnequalExamples(List<T> value) {
-        return new Configuration<>(type, prefabValues, equalExamples, value, ignoredFields, nonnullFields, ignoredAnnotations,
-                cachedHashCodeInitializer, hasRedefinedSuperclass, redefinedSubclass, usingGetClass, warningsToSuppress);
+        return new Configuration<>(type, prefabValues, equalExamples, value, actualFields, excludedFields,
+                includedFields, nonnullFields, ignoredAnnotations, cachedHashCodeInitializer, hasRedefinedSuperclass,
+                redefinedSubclass, usingGetClass, warningsToSuppress);
     }
 
     public List<T> getUnequalExamples() {
         return Collections.unmodifiableList(unequalExamples);
     }
 
-    public Configuration<T> withIgnoredFields(List<String> value) {
-        return new Configuration<>(type, prefabValues, equalExamples, unequalExamples, new HashSet<>(value), nonnullFields, ignoredAnnotations,
-                cachedHashCodeInitializer, hasRedefinedSuperclass, redefinedSubclass, usingGetClass, warningsToSuppress);
+    public Set<String> getActualFields() {
+        return Collections.unmodifiableSet(actualFields);
     }
 
+    public Set<String> getExcludedFields() {
+        return Collections.unmodifiableSet(excludedFields);
+    }
+
+    public Configuration<T> withExcludedFields(List<String> value) {
+        return new Configuration<>(type, prefabValues, equalExamples, unequalExamples, actualFields, new HashSet<>(value),
+                includedFields, nonnullFields, ignoredAnnotations, cachedHashCodeInitializer, hasRedefinedSuperclass,
+                redefinedSubclass, usingGetClass, warningsToSuppress);
+    }
+
+    public Set<String> getIncludedFields() {
+        return Collections.unmodifiableSet(includedFields);
+    }
+
+    public Configuration<T> withIncludedFields(List<String> value) {
+        return new Configuration<>(type, prefabValues, equalExamples, unequalExamples, actualFields, excludedFields,
+                new HashSet<>(value), nonnullFields, ignoredAnnotations, cachedHashCodeInitializer, hasRedefinedSuperclass,
+                redefinedSubclass, usingGetClass, warningsToSuppress);
+    }
 
     public Set<String> getIgnoredFields() {
-        return Collections.unmodifiableSet(ignoredFields);
+        Set<String> ignoredFields = new HashSet<>();
+        if (excludedFields.isEmpty() && !includedFields.isEmpty()) {
```

Comment:
Let's reduce cyclomatic complexity by inverting the condition and returning the `ignoredFields` immediately, and then de-indent the for loop.